### ISSUE TITLE
Workbench: Indicate to the user if a newer version is available

### ISSUE THREAD

**Issue by davemfish** [natcap/invest#1369]
We implemented this in Qt, so it could be useful to remind ourselves how that worked

**Comment by emilyanndavis** [comment]
[Electron's guide to Updating Applications](https://www.electronjs.org/docs/latest/tutorial/updates) outlines a few options that allow an app to automatically check for—and download—a new version. From what I can gather, key options (with annoyingly similar names) include the following:
- [`autoUpdater`](https://www.electronjs.org/docs/latest/api/auto-updater), the officially supported API built into Electron
- [`update-electron-app`](https://www.electronjs.org/docs/latest/tutorial/updates#using-updateelectronjsorg), also maintained by the Electron team, which abstracts away some configuration and is designed to work with GitHub Releases
- [`electron-updater`](https://www.electron.build/auto-update), a configuration-driven solution designed to work with `electron-builder`

One thing all these options have in common is that they _automatically download_ an update once it's available. It doesn't appear to be possible to simply check for an update without triggering a download. While there's a compelling argument for reducing friction when it comes to keeping users' software up to date in the case of apps with huge user bases and/or with high potential for security risks, that's not necessarily what we need to do here. If we want a bare-bones, in-app notification that a new version is available, otherwise giving users full control over the update process itself, we can certainly come up with a custom solution.

If we do want to go the auto-update route, it's not immediately clear to me how these drop-in solutions would interact with our current practice of including the version number in the product name. These auto-updaters seem to assume a user has a single instance of an application installed, and when a new version is installed (the next time the app is restarted after downloading a new version), it replaces the existing version. We'll need to decide whether we want to
1. switch to a single-instance, always-up-to-date model (in which case we should be able to use a drop-in auto-update solution),
2. continue supporting multiple installations, as many as one per bugfix version (in which case we might be able to configure a drop-in solution for auto-download, but users would most likely have to manually install & run the automatically downloaded version), or
3. maybe even implement something in-between (say, for example, supporting co-existing minor versions while auto-updating for every bugfix version).

**Comment by davemfish** [comment]
I think one concern about auto-updating users is it could inhibit reproducibility. Since this is scientific software it's pretty important that results can be reproduced exactly, if someone wants to do that. So while we definitely want folks to always use the latest version, there is a use-case for needing to run a specific older version.

**Comment by emilyanndavis** [comment]
A straightforward way to determine whether a new version is available (without auto-updating) would be to [fetch information about the latest release using the GitHub REST API](https://docs.github.com/en/rest/releases/releases?apiVersion=2022-11-28#get-the-latest-release). Since the `invest` repo is public, no authentication is needed (I tested this and can confirm). The response object includes the newest version number, which we can easily compare to the user's list of active versions to determine whether they already have the latest or not.

Some implementation & UX details to think about:
- when/how frequently to query the API.
- whether to use [conditional requests](https://docs.github.com/en/rest/using-the-rest-api/best-practices-for-using-the-rest-api?apiVersion=2022-11-28#use-conditional-requests-if-appropriate) based on when the latest release info was last modified.
- what information to include in the notification, and whether to include links (e.g., InVEST download page, changelog, etc.).
- what actions (f any) to provide alongside the notification (e.g., download new version now, remind me later, etc.).
- whether/how frequently to remind users who have been notified but have not yet updated.

**Comment by davemfish** [comment]
I agree, the QGIS style alert with a link to a download page seems appropriate. I think I'm okay with showing it every time someone opens the Workbench, so long as it is an unobtrusive location. Probably somewhere on the Home tab.

And fetching to find out if there is a newer release sounds like a great idea. PyPI is another option if that's preferable to GitHub for some reason. Both are equally valid sources of info, I think. Here's how we used to do it for the old Qt UI:

```python
    def _get_latest_version():
        """Get the latest InVEST version string from PyPI page.

        Returns:
            latest_version (str): if the HTTP request is successfully, or
                None if not.

        """
        # Make an HTTP call to InVEST's PyPI page, set timeout of 10s
        try:
            response = requests.get(
                'https://pypi.python.org/pypi/natcap.invest/json', timeout=10)
            # Get the latest version string
            latest_version = json.loads(response.text)['info']['version']
            return latest_version

        # If any ConnectionError, HTTPError, Timeout, or TooManyRedirects
        # exception happens
        except requests.exceptions.RequestException as err:
            LOGGER.exception('Exception while requesting PyPI page: %s' % err)

        # If the text doesn't have the 'info' or 'version' keys
        except KeyError as err:
            LOGGER.exception('Version string could not be found from PyPI page.'
                             'Exception raised: %s' % err)
        return None
```
